File: rover/ros2/src/high_level_nav2_commander_py/high_level_nav2_commander_py/high_level_commander_node.py
import rclpy
from rclpy.node import Node
from rclpy.executors import MultiThreadedExecutor


class Nav2CommanderNode(Node):
    def __init__(self, node_name):
        Node.__init__(self, node_name=node_name)
        self.get_logger().info("Hi! I'm the nav2 commander node!")


def main():
    rclpy.init()
    nav2_commander = Nav2CommanderNode("nav2_commander_py")

    executor = MultiThreadedExecutor()
    executor.add_node(nav2_commander)
    executor.spin()

    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: rover/ros2/src/high_level_nav2_commander_cpp/src/high_level_commander_node.cpp
#include "high_level_nav2_commander_cpp/high_level_commander.hpp"

Nav2CommanderNode::Nav2CommanderNode(const std::string& node_name) : Node(node_name, rclcpp::NodeOptions())
{
  std::cout << "Hi! I'm the nav2 commander node!" << std::endl;
}

int main(int argc, char* argv[])
{
  rclcpp::init(argc, argv);
  auto nav2_commander = std::make_shared<Nav2CommanderNode>("nav2_commander_cpp");

  rclcpp::executors::MultiThreadedExecutor executor;
  executor.add_node(nav2_commander);
  executor.spin();

  rclcpp::shutdown();

  return 0;
}